packages/protoplugin/src/ecmascript/schema.ts

```diff
@@ -132,17 +133,37 @@ export function createSchema(
   };
   return {
     schema,
-    toResponse(res) {
-      res.supportedFeatures = protoInt64.parse(
-        CodeGeneratorResponse_Feature.PROTO3_OPTIONAL
-      );
-      for (const genFile of generatedFiles) {
-        genFile.toResponse(res);
-      }
+    getFileInfo() {
+      // TODO - If all targets are requested, but the plugin doesn't have a generateDts, we might need to
+      // retrieve only the ts files here. If that turns out to be the case, we can change the method behavior
+      // to evict the generated files. As in "get the files generated so far, and remove them from
+      // the schema controller".
```

Comment:
Well things work, but we still see those pre-emit diagnostics error messages, such as:

`"File 'proto/address_book_proto.js' is a JavaScript file. Did you mean to enable the 'allowJs' option?",`

and

`"Cannot redeclare block-scoped variable 'test'."`

which are both related to us passing the `.js` files to the transpiler.  I think ideally we'd like to only send `.ts` files here.  I can't think of a scenario where we'd ever want to pass `.js` files.  We require a `generateTs` function, so users would never be able to only specify a `generateJs` function and expect to transpile declarations.

What are your thoughts?  If we can't think of a reason why a user would want to send JS through the transpiler, then I think I should just prevent that from happening here.  In looking at it, it would be pretty easy.  I'll just save off the results of `getFileInfo` after the call to `generateTs` and only send that array to the transpiler if needed.

Reply:
> I'll just save off the results of getFileInfo after the call to generateTs and only send that array to the transpiler if needed.

Sounds like a good plan :+1: